test/unit/acknowledgement/process-invalid.test.js

```diff
@@ -0,0 +1,242 @@
+const mockCommit = jest.fn()
+const mockRollback = jest.fn()
+const mockTransaction = {
+  commit: mockCommit,
+  rollback: mockRollback
+}
+
+jest.mock('../../../app/data', () => {
+  return {
+    sequelize:
+       {
+         transaction: jest.fn().mockImplementation(() => {
+           return { ...mockTransaction }
+         })
+       }
+  }
+})
+
+jest.mock('../../../app/config')
+const mockConfig = require('../../../app/config')
+
+jest.mock('../../../app/event')
+const mockEvent = require('../../../app/event')
+
+jest.mock('../../../app/acknowledgement/get-hold-category-name')
+const getHoldCategoryName = require('../../../app/acknowledgement/get-hold-category-name')
+
+jest.mock('../../../app/holds/get-hold-category-id')
+const getHoldCategoryId = require('../../../app/holds/get-hold-category-id')
+
+jest.mock('../../../app/reschedule/index')
+const holdAndReschedule = require('../../../app/reschedule/index')
+
+jest.mock('../../../app/reset/index')
+const { resetPaymentRequestById } = require('../../../app/reset/index')
+
+const mockFRN = require('../../mock-frn')
+const processInvalid = require('../../../app/acknowledgement/process-invalid')
+const { SFI } = require('../../../app/schemes')
+
+let mockAcknowledgement
+let mockAcknowledgementError
+let mockHoldCategoryName
+let mockHoldCategoryId
+let mockHoldReschedule
+let mockResetPaymentRequestById
+
+let schemeId
+let paymentRequestId
+
+describe('send acknowledgement error event', () => {
+  beforeEach(() => {
+    mockConfig.isAlerting = true
+
+    mockAcknowledgement = JSON.parse(JSON.stringify(require('../../mock-acknowledgement')))
+    mockAcknowledgementError = JSON.parse(JSON.stringify(require('../../mock-acknowledgement-error')))
+    mockHoldCategoryName = JSON.parse(JSON.stringify(require('../../constants/hold-categories-names')))
+
+    schemeId = SFI
+    paymentRequestId = 1
+
+    getHoldCategoryName.mockReturnValue(
+      mockHoldCategoryName
+    )
+
+    getHoldCategoryId.mockReturnValue(
+      mockHoldCategoryId
+    )
+
+    holdAndReschedule.mockReturnValue({
+      mockHoldReschedule
+    })
+
+    resetPaymentRequestById.mockReturnValue({
+      mockResetPaymentRequestById
+    })
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should return undefined when processInvalid is called', async () => {
+    const result = await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError)
+    expect(result).toBeUndefined()
+  })
+
+  test('should call resetPaymentRequestById when a schemeId, paymentRequestId, frn and transaction are given', async () => {
+    await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError)
+    expect(resetPaymentRequestById).toHaveBeenCalled()
+  })
+
+  test('should call resetPaymentRequestById once when a schemeId, paymentRequestId, frn and transaction are given', async () => {
```

Comment:
test names are a little bit off -- mainly due to copy-pasta.

what I would suggest is this, we have our initial situation set:

- schemeId of SFI
- paymentRequestId of 1
- FRN of 1234567890
- acknowledgeError as defined in its file

This is the default case.

For this, I wouldn't specify what you've called the `processInvalid` with because this is derived from our initial state, e.g. line 113 would just become `test('should call getHoldCategoryName', async () => {`.

What I would call out in later tests is when you change these default inputs, i.e. test starting line 168, we've changed the `mockConfig.isAlerting` from our initial state.

I think we should make better use of `describe` to help split out our test files based on situations -- but, hey oh!